components/datetime/src/fields/symbols.rs

```diff
@@ -676,54 +675,36 @@ field_type!(
         /// Field symbol for the specific non-location format of a time zone.
         ///
         /// For example: "Pacific Standard Time"
-        'z' => LowerZ = 0,
-        /// Field symbol for any of: the ISO8601 basic format with hours, minutes and optional seconds fields, the
-        /// long localized offset format, or the ISO8601 extended format with hours, minutes and optional seconds fields.
-        'Z' => UpperZ = 1,
+        'z' => SpecificNonLocation = 0,
         /// Field symbol for the localized offset format of a time zone.
         ///
         /// For example: "GMT-07:00"
-        'O' => UpperO = 2,
+        'O' => LocalizedOffset = 1,
         /// Field symbol for the generic non-location format of a time zone.
         ///
         /// For example: "Pacific Time"
-        'v' => LowerV = 3,
+        'v' => GenericNonLocation = 2,
         /// Field symbol for any of: the time zone id, time zone exemplar city, or generic location format.
-        'V' => UpperV = 4,
-        /// Field symbol for either the ISO8601 basic format or ISO8601 extended format, with an optional ISO8601 UTC indicator `Z`.
-        'x' => LowerX = 5,
-        /// Field symbol for either the ISO8601 basic format or ISO8601 extended format.  This does not allow an
-        /// optional ISO8601 UTC indicator `Z`, whereas [`TimeZone::LowerX`] allows the optional `Z`.
-        'X' => UpperX = 6,
+        'V' => Location = 3,
+        /// Field symbol for either the ISO-8601 basic format or ISO-8601 extended format. This does not use an
+        /// optional ISO-8601 UTC indicator `Z`, whereas [`TimeZone::IsoWithZ`] produces `Z`.
+        'x' => Iso = 4,
+        /// Field symbol for either the ISO-8601 basic format or ISO-8601 extended format, with the ISO-8601 UTC indicator `Z`.
+        'X' => IsoWithZ = 5,
     };
```

Comment:
I don't know what @sffc and @Manishearth discussed before, but after this PR, custom behaviour could be added like this:

```suggestion
    _ => WhateverChromeDoes = 7,
    };
```

This variant can be constructed as `TimeZone::WhateverChromeDoes`, but doesn't have an associated char for parsing.